### PR TITLE
[OFFAPPS-978] hide empty fields

### PR DESCRIPTION
**Pull request by ocke**
WIP.

Hide empty fields.

Like requested here: https://github.com/zendesk/user_data_app/issues/80

### CODE REVIEW

**Comment by sostopher** [comment]
Let's pick this up next Sprint in that case @ocke

**Comment by sostopher** [comment]
@ocke Please make sure we separate the PR for the new translation string.

File: src/javascript/app.js [hunk 1]
```diff
@@ -230,6 +232,8 @@ const app = {
         result.simpleKey = ['custom', key].join(' ')
         result.value = values[key]
 
+        if (setting('hideEmptyFields') && !result.value && !result.editable) { return null }
```
Comment (yu-tian113) [inline src/javascript/app.js]: this line shows up a couple times, can we move it in a function or variable?

File: src/main.scss [hunk 2]
```diff
@@ -35,7 +35,7 @@ html, body {
     margin-bottom: 1em;
   }
 
-  .org_fields {
+  div.hide_empty_fields, .org_fields {
     margin-top: 20px;
```
Comment (yu-tian113) [inline src/main.scss]: there should be a utility class for margin if you already import css-utility

File: src/javascript/app.js (hunk 1 above)
Comment (vaevictis) [reply src/javascript/app.js]: Agreed with @yu-tian113

File: src/main.scss [hunk 3]
```diff
@@ -35,7 +35,7 @@ html, body {
     margin-bottom: 1em;
   }
 
-  .org_fields {
+  div.hide_empty_fields, .org_fields {
```
Comment (vaevictis) [inline src/main.scss]: Not sure the `div` selector is useful on top of the class.

File: src/main.scss (hunk 2 above)
Comment (ocke) [reply src/main.scss]: I don't import the utility

File: src/main.scss [hunk 4]
```diff
@@ -82,22 +82,33 @@ i.icon_circle_arrow_left {
...
   }
   &.closed {
-    background: #DDDDDD;
+    background: #D8DCDE;
+    color: #2F3941;
+  }
+  &.disabled {
+    background: 'transparent';
+    color: #C2C8CC;
+    borderColor: #D8DCDE;
```
Comment (sostopher) [inline src/main.scss]: @ocke is this an issue?

**Pull request merged**